File: maven-confluence-core/src/test/java/org/bsc/confluence/ConfluenceUtilsTest.java
package org.bsc.confluence;

import org.hamcrest.core.IsEqual;
import org.junit.Assert;
import org.junit.Ignore;
import org.junit.Test;

import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

import static org.bsc.confluence.ConfluenceHtmlUtils.replaceHTML;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.*;

public class ConfluenceUtilsTest {

    @Test
    public void decode() throws Exception {
        String input = "<coDe>[I][RE]'.*moon[^dab]+'</CODE> : <EM>all</Em> <u>implemented</U> <s>specifications</S> <DEL>having</del> <Strong>the</Strong> <i>RE<I> '.*moon[^dab]+'<BR/>";
        String decoded = replaceHTML(input);
        assertEquals("{{\\[I\\]\\[RE\\]'.\\*moon\\[^dab\\]\\+'}} : _all_ +implemented+ -specifications- -having- *the* _RE_ '.\\*moon\\[^dab\\]\\+'\\\\", decoded);
    }

    @Test
    @Ignore
    public void pathTest() throws Exception {

        //File f = new File(".");

        Path pp = Paths.get(  "http://test.xml");

        Assert.assertThat( Files.exists(pp), IsEqual.equalTo(false) );

        Path p1 = Paths.get(  "classpath:test.xml");

        Assert.assertThat( Files.exists(p1), IsEqual.equalTo(false) );

    }

    @Test
    public void sanitationNotNeeded() {
        assertThat(ConfluenceUtils.sanitizeLabel("label"), is("label"));
    }

    @Test
    public void sanitation() {
        assertThat(ConfluenceUtils.sanitizeLabel("labe:l"), is("labe-l"));
    }

    @Test
    public void sanitationFull() {
        assertThat(ConfluenceUtils.sanitizeLabel(": ; , . , ? & [ ] ( ) # ^ * ! @"),
                is("-------------------------------"));
    }

}

File: maven-confluence-core/src/test/kotlin/org/bsc/confluence/ConfluenceUtilsTest.kt
package org.bsc.confluence

import org.bsc.confluence.ConfluenceHtmlUtils.replaceHTML
import org.bsc.confluence.ConfluenceUtils.sanitizeLabel
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
import java.nio.file.Files
import java.nio.file.Paths


class ConfluenceUtilsTest {
    @Test
    @Throws(Exception::class)
    fun decode() {
        val input = "<coDe>[I][RE]'.*moon[^dab]+'</CODE> : <EM>all</Em> <u>implemented</U> <s>specifications</S> <DEL>having</del> <Strong>the</Strong> <i>RE<I> '.*moon[^dab]+'<BR/>"
        val decoded = replaceHTML(input)
        assertEquals("{{\\[I\\]\\[RE\\]'.\\*moon\\[^dab\\]\\+'}} : _all_ +implemented+ -specifications- -having- *the* _RE_ '.\\*moon\\[^dab\\]\\+'\\\\", decoded)
    }

    @Test
    @Disabled
    @Throws(Exception::class)
    fun pathTest() {

        //File f = new File(".");
        val pp = Paths.get("http://test.xml")
        assertFalse(Files.exists(pp))
        val p1 = Paths.get("classpath:test.xml")
        assertFalse(Files.exists(p1))
    }

    @Test
    fun sanitationNotNeeded() = assertEquals( "label", sanitizeLabel("label"))

    @Test
    fun sanitation() = assertEquals( "labe-l", sanitizeLabel("labe:l"))

    @Test
    fun sanitationFull() = assertEquals( "-------------------------------", sanitizeLabel(": ; , . , ? & [ ] ( ) # ^ * ! @") )
}